pkg/envoy/sds/builder.go

```diff
@@ -0,0 +1,163 @@
+package sds
+
+import (
+	xds_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	xds_auth "github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/tls/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/envoy/secrets"
+	"github.com/openservicemesh/osm/pkg/errcode"
+	"github.com/openservicemesh/osm/pkg/identity"
+	"github.com/openservicemesh/osm/pkg/service"
+)
+
+type sdsBuilder struct {
+	proxy *envoy.Proxy
+
+	// Service certificate for this proxy
+	proxyCert *certificate.Certificate
+
+	// Requested certs to generate secrets for
+	requestedCerts []string
+
+	trustDomain string
+
+	serviceIdentitiesForServices map[service.MeshService][]identity.ServiceIdentity
+}
+
+func newSdsBuilder() *sdsBuilder {
+	return &sdsBuilder{}
+}
+
+func (b *sdsBuilder) SetProxy(proxy *envoy.Proxy) {
+	b.proxy = proxy
+}
```

Comment:
Why is `newSdsBuilder()` unexported while the methods on the builder type are exported?